### DISCUSSION TITLE
How to debug?

### DISCUSSION THREAD

**Discussion by dclipca** [99designs/gqlgen#2001] [Q&A]
How do I debug where a resolver returns and so on? Is there a guide I can look up?

**Comment by frederikhors** [comment]
You can use the Golang tools for debugging, nope?

**Reply by dclipca** [reply]
Yes that's true but I would have to put a log at every return, no?

I wonder if there is some kind of middleware or function that I can attach my logger to gqlgen so that I will not have to copy-paste `log.Println()` one thousand times.

**Reply by dclipca** [reply]
My usecase is determining at what place in my code the resolver returns with an error.
